app/web/features/search/SearchPage.tsx

```diff
@@ -52,193 +55,195 @@ const useStyles = makeStyles((theme) => ({
-      setSelectedResult({ username, userId, lng, lat });
-    },
-    []
-  );
-
-  //detect when map data has been initially loaded
-  const handleMapSourceData = useCallback(() => {
-    if (
-      map.current &&
-      map.current.getSource("clustered-users") &&
-      map.current.isSourceLoaded("clustered-users")
-    ) {
-      setAreClustersLoaded(true);
-
-      // unbind the event
-      map.current.off("sourcedata", handleMapSourceData);
-    }
-  }, []);
-
-  useEffect(() => {
-    if (!map.current) return;
-    const handleMapClickAway = (e: EventData) => {
-      //defaultPrevented is true when a map feature has been clicked
-      if (!e.defaultPrevented) {
-        setSelectedResult(undefined);
+        return service.search.userSearch(
+          {
+            query: queryName,
+            bbox: locationResult.bbox,
+            lastActive:
+              lastActiveComparation === 0 ? undefined : lastActiveFilter,
+            hostingStatusOptions:
+              hostingStatusFilterComparation === 0
+                ? undefined
+                : [hostingStatusFilter],
+            numGuests: numberOfGuestFilter,
+            completeProfile:
+              completeProfileFilter === false
+                ? undefined
+                : completeProfileFilter,
+          },
+          pageParam
+        );
+      },
+      {
+        getNextPageParam: (lastPage) =>
+          lastPage.nextPageToken ? lastPage.nextPageToken : undefined,
       }
-    };
-
-    //bind event handlers for map events (order matters!)
-    map.current.on(
-      "click",
-      layers.unclusteredPointLayer.id,
-      handleMapUserClick
     );
-    map.current.on("click", handleMapClickAway);
 
-    map.current.on("sourcedata", handleMapSourceData);
-
-    return () => {
-      if (!map.current) return;
+  // Relocate map everytime boundingbox changes
+  useEffect(() => {
+    map.current?.fitBounds(locationResult.bbox, {
+      maxZoom: selectedUserZoom,
+    });
+  }, [locationResult.bbox]);
 
-      //unbind event handlers for map events
-      map.current.off("sourcedata", handleMapSourceData);
-      map.current.off("click", handleMapClickAway);
-      map.current.off(
-        "click",
-        layers.unclusteredPointLayer.id,
-        handleMapUserClick
+  // Initial bounding box
+  useEffect(() => {
+    if (isLoadingUser === false && userData && bbox.join() === "0,0,0,0") {
+      map.current?.fitBounds(
+        [userData.lng, userData.lat, userData.lng, userData.lat],
+        {
+          maxZoom: 4,
+        }
       );
-    };
-  }, [handleMapUserClick, handleMapSourceData]);
 
-  const initializeMap = (newMap: MaplibreMap) => {
-    map.current = newMap;
-    newMap.on("load", () => {
-      addClusteredUsersToMap(newMap);
-    });
-  };
+      map.current?.fitBounds(map.current?.getBounds());
+      setMapInitiallyLocated(true);
+    }
+  }, [isLoadingUser]);
+
+  let errorMessage = error?.message || undefined;
```

Comment:
You don't need the `undefined` here. The expression `error?.message` will return `undefined` if error is `null` or `undefined`, or if `error.message` itself is `undefined`.

Reply:
done ✅